### ISSUE TITLE
Extension does not detect existing site and adds a new one instead of updating it

### ISSUE THREAD

**Issue by jpbaril** [bitwarden/browser#311]
I had in my vault a site which only had a password (no username because I created the entry after using the site reset password feature). After logging out and back in Bitwarden Firefox 56 extension asked me to add the site to my vault. I thought it was maybe just a string error and that it would simply update my existing entry with the username. No, it added a second site in my vault.

**Comment by kspearrin** [comment]
How are we to know that the account in your vault is the one you just logged into if there is no username associated with it?

**Comment by jpbaril** [comment]
Well, the URI was the same (and the password also, but that's not relevant). The extension could have just asked me if I wanted to update the existing site or if I wanted to create a new one. That's what LastPass do.

**Comment by kspearrin** [comment]
LastPass will prompt you **every time** you log in somewhere when your vault is locked? Seems like that would get very annoying.

**Comment by kspearrin** [comment]
Sorry, I got your comment confused with another thread. We can investigate adding an update option to the save button.

**Comment by jpbaril** [comment]
Not sure to understand what you mean.

BTW, on my home desktop pc, my vault is always unlocked, if that's any indication.

What LastPass does after logging in a site is to detect that a site with the same URI as the one visited exists in the vault and propose to update it or create a new one. Bitwarden could probably do that as well as it correctly found the site in my vault and filled the password in the logging form. After logging in by manually filling the missing username, Bitwarden ask me to add the site as a new one. I mean, if it detected and filled the form just earlier with that specific entry, it obviously assumed it was the good credentials to use. Now that I add an information that was missing in the entry, why not ask me if I would like to update the entry in the vault?

**Comment by jyavenard** [comment]
@kspearrin
> LastPass will prompt you every time you log in somewhere when your vault is locked? Seems like that would get very annoying.

no, last pass ask you if you want to update the current entry or enter a new one.

this is the proper way to do it.
